nomenclature/processor/data_validator.py

```diff
@@ -175,6 +189,33 @@ def filter_args(self):
     def __str__(self):
         return ", ".join([f"{key}: {value}" for key, value in self.filter_args.items()])
 
+    def apply(
```

Comment:
I'd strongly suggest against using `fail_list` and `output_list` as input values that we append to and not explicitly return. This is programming by side effect and is not recommended since it makes it harder to understand what is going on.
Instead I'd recommend to change the function to either return a tuple of a dictionary that contains `[error, fail_list, output_list]` and work with it further down the line.